EarlGrey/Additions/UIViewController+GREYAdditions.m

```diff
@@ -138,14 +138,18 @@ - (void)greyswizzled_viewWillAppear:(BOOL)animated {
       // Interactive transitions can cancel and cause imbalance of will and did calls.
       id<UIViewControllerTransitionCoordinator> coordinator = [self transitionCoordinator];
       if (coordinator && [coordinator initiallyInteractive]) {
-        [coordinator notifyWhenInteractionEndsUsingBlock:
-         ^(id<UIViewControllerTransitionCoordinatorContext> context) {
-           if ([context isCancelled]) {
-             id object =
-                 objc_getAssociatedObject(self, @selector(greyswizzled_viewWillAppear:));
-             UNTRACK_STATE_FOR_OBJECT(kGREYPendingViewsToAppear, object);
-           }
-         }];
+        id block = ^(id<UIViewControllerTransitionCoordinatorContext> context) {
+          if ([context isCancelled]) {
+            id object =
+                objc_getAssociatedObject(self, @selector(greyswizzled_viewWillAppear:));
+            UNTRACK_STATE_FOR_OBJECT(kGREYPendingViewsToAppear, object);
+          }
+        };
+#if !defined(__IPHONE_10_0) || __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_10_0
```

Comment:
You could also use iOS10_OR_ABOVE() from the defines.

https://github.com/google/EarlGrey/blob/master/EarlGrey/Common/GREYDefines.h#L35

Reply:
Same throughout.

Reply:
Will that work? I thought that would just check the current runtime, and not the deployment target being used.

Reply:
Oh, you're right on that. Just realized we don't have any defines for compile time checks. 

If repeated, feel free to add them in GREYDefines or so, otherwise, this looks good to me.